src/Microsoft.AspNet.Mvc.Core/Rendering/Html/HtmlHelper.cs

```diff
@@ -534,19 +534,26 @@ public static string GetFormMethodString(FormMethod method)
         /// <inheritdoc />
         public HtmlString TextArea(string expression, string value, int rows, int columns, object htmlAttributes)
         {
-            var metadata = ExpressionMetadataProvider.FromStringExpression(expression, ViewData, MetadataProvider);
+            var modelExplorer = ExpressionMetadataProvider.FromStringExpression(expression, ViewData, MetadataProvider);
             if (value != null)
             {
-                metadata.Model = value;
+                // As a special case we allow treating a string value as a model of arbitrary type.
+                // So pass through the string representation, even though the modelmetadata might
+                // be for some other type.
```

Comment:
nits:
- ModelMetadata
- mention _why_ we allow this special case.  e.g. something about generating both the given `value` and correct validation attributes even if the `value`  is the result of a conversion (or simply `ToString()`) from some other type.